### PR TITLE
doc: add algorithms overview and decen sgd intro

### PR DESCRIPTION
**Pull request by ganshaoduo**

### CODE REVIEW

**Comment by NOBLES5E** [comment]
Don't push .DS_Store files

File: src/algorithms/decentralized.md [hunk 1]
```diff
@@ -2,41 +2,40 @@
 # Decentralized SGD
 
 
-## Overview
-Decentralized SGD is a data-parallel distributed learning algorithm that removes the requirement of a centralized global model among all workers, which makes it quite different from Allreduce-based or Parameter Server-based algorithms regarding the communication pattern. With decentralized SGD, each worker only needs to exchange data with one or a few specific workers, instead of aggregating data globally. Therefore, decentralized communication has much less communication connections than Allreduce, and more balanced communication overhead than Parameter Server. Although decentralized SGD may lead to different models on each worker, it has been proved in theory that the convergence rate of the decentralized SGD is same as its centralized counterpart. You can find the detailed analysis about decentralized SGD in our [paper](https://arxiv.org/abs/1705.09056).
+## Overview of Decentralized Training
```
Comment (NOBLES5E) [inline src/algorithms/decentralized.md:5]: only capitalize the first work like other files

File: src/algorithms/decentralized.md [hunk 2]
```diff
@@ -2,41 +2,40 @@
...
+| Allreduce (Ring)      | $\mathcal{O}(n)$ | $\mathcal{O}(1)$ |
+| Parameter Server      | $\mathcal{O}(1)$ | $\mathcal{O}(n)$ |
+| Decentralized SGD in Bagua | $\mathcal{O}(1)$ | $\mathcal{O}(1)$ |
 
 
+## Example
 
 
-## Example
+## Benchmark
```
Comment (NOBLES5E) [inline src/algorithms/decentralized.md:38]: which do you think better? leave benchmark in a dedicated page or in each algorithm page?

**Review by NOBLES5E — CHANGES_REQUESTED** [review]
see comments

File: src/algorithms/index.md [hunk 3]
```diff
@@ -1 +1,6 @@
 # Algorithms
+
+
+Bagua thrives on the diversity of distributed learning algorithms. The great flexibility of the system makes it possible to smoothly incoporate various of SOTA algorithms while providing automatic optimizations for the performance during the execution. For the end user, Bagua provides a wide range of choices of algorithms, which she can easily try out for her tasks. For the algorithm developer, Bagua is a playground where she can only be focused on the algorithm itself (e.g., the logic and control) without reinventing the wheels (e.g., communication primitives and system optimizations) across different algorithms.
+
+In this tutorial, we will describe several algorithms that have already been implemented within Bagua, including the main ideas of each algorithm and their usage in specific examples. Then we are going to demonstrate how to add a new algotirhm into Bagua.
```
Comment (NOBLES5E) [inline src/algorithms/index.md:6]: typo (algorithm)

File: src/algorithms/decentralized.md (hunk 2 above)
Comment (ganshaoduo) [reply src/algorithms/decentralized.md:38]: I would prefer a dedicated page for all the benchmarks? Then the tutorial can be focused on the explanation and usage of the algorithm.

**Review by ganshaoduo — COMMENTED** [review]
typo fixed, title fixed.
see comment about the benchmark.

File: src/algorithms/decentralized.md (hunk 2 above)
Comment (NOBLES5E) [reply src/algorithms/decentralized.md:38]: Then this section contains link to the corresponding section in benchmark page?

Comment (ganshaoduo) [reply src/algorithms/decentralized.md:38]: sure, but I assume we will have one page for all benchmarks, and some benchmarks involved multiple algorithms, then maybe we won't have independent sections for each algorithm. Another option is we just put the benchmark page in a parallel directory as the Algorithm, then it's quite clear to me.

**Pull request merged**